### ISSUE TITLE
[ERROR][com.freerdp.client.x11] - RAIL exec error: execResult=RAIL_EXEC_E_FAIL NtError=0x2

### ISSUE THREAD

**Issue by rodrigo-yoshioka** [FreeRDP/FreeRDP#3841]
Hi, 
I've installed FreeRDP from binaries and from GIT compiling it from source. In both ways I used a Fedora 25 with KDE environment, and I having a issue when trying to use remoteapp to open applications in a Windows 7 Ultimate OS. I'll try to send all information you may need to help me.

```
xfreerdp -u Rodrigo -p PASWD /cert-ignore --app --plugin rail --data '||Calculator' -- 192.168.122.133
[21:30:18:761] [11688:11688] [WARN][com.freerdp.client.common.cmdline] - Using deprecated command-line interface!
[21:30:18:762] [11688:11688] [WARN][com.freerdp.client.common.compatibility] - -p ****** -> /p:******
[21:30:18:762] [11688:11688] [WARN][com.freerdp.client.common.compatibility] - -u Rodrigo -> /u:Rodrigo
[21:30:18:762] [11688:11688] [WARN][com.freerdp.client.common.compatibility] - --app -> /app: + program name or alias
[21:30:18:762] [11688:11688] [WARN][com.freerdp.client.common.compatibility] - 192.168.122.133 -> /v:192.168.122.133
[21:30:18:762] [11688:11688] [WARN][com.freerdp.client.common.compatibility] - 
[21:30:18:765] [11688:11689] [INFO][com.freerdp.client.common.cmdline] - loading channelEx rail
[21:30:18:950] [11688:11689] [INFO][com.freerdp.gdi] - Local framebuffer format  PIXEL_FORMAT_BGRX32
[21:30:18:950] [11688:11689] [INFO][com.freerdp.gdi] - Remote framebuffer format PIXEL_FORMAT_RGB16
[21:30:19:086] [11688:11694] [ERROR][com.freerdp.client.x11] - RAIL exec error: execResult=RAIL_EXEC_E_FAIL NtError=0x2
```

```
uname -r
4.9.12-200.fc25.x86_64
```

```
xfreerdp --version
This is FreeRDP version 2.0.0-dev (git f77b4a5)
```

When I remove "App" part, I'm able to open Remote Desktop.

![screenshot_20170303_214253](https://cloud.githubusercontent.com/assets/22940874/23574170/88fad73e-005a-11e7-8870-16fc308cd24e.png)

I've looking on google about this issue, but everything I found didn't solve my problem.

Thanks.

**Comment by hardening** [comment]
This has to do with your windows configuration. You could try to give the full path to the executable.

**Comment by rodrigo-yoshioka** [comment]
Hi, 
I tried with full path, but now the errors are different...

```
[root@localhost rodrigo]# xfreerdp -u Rodrigo -p PASSWD /cert-ignore --app --plugin rail --data '%windir%\system32\calc.exe' -- 192.168.122.133
[07:36:06:626] [22397:22397] [WARN][com.freerdp.client.common.cmdline] - Using deprecated command-line interface!
[07:36:06:626] [22397:22397] [WARN][com.freerdp.client.common.compatibility] - -p ****** -> /p:******
[07:36:06:626] [22397:22397] [WARN][com.freerdp.client.common.compatibility] - -u Rodrigo -> /u:Rodrigo
[07:36:06:626] [22397:22397] [WARN][com.freerdp.client.common.compatibility] - --app -> /app: + program name or alias
[07:36:06:626] [22397:22397] [WARN][com.freerdp.client.common.compatibility] - 192.168.122.133 -> /v:192.168.122.133
[07:36:06:626] [22397:22397] [WARN][com.freerdp.client.common.compatibility] - 
[07:36:06:626] [22397:22398] [INFO][com.freerdp.client.common.cmdline] - loading channelEx rail
[07:36:07:131] [22397:22398] [INFO][com.freerdp.gdi] - Local framebuffer format  PIXEL_FORMAT_BGRX32
[07:36:07:131] [22397:22398] [INFO][com.freerdp.gdi] - Remote framebuffer format PIXEL_FORMAT_RGB16
[07:36:51:692] [22397:22398] [ERROR][com.freerdp.core.fastpath] - FASTPATH_UPDATETYPE_ORDERS - fastpath_recv_orders()
[07:36:51:692] [22397:22398] [ERROR][com.freerdp.core.fastpath] - fastpath_recv_update() - -1
[07:36:51:692] [22397:22398] [ERROR][com.freerdp.core.fastpath] - fastpath_recv_update_data() fail
[07:36:51:692] [22397:22398] [ERROR][com.freerdp.core.transport] - transport_check_fds: transport->ReceiveCallback() - -1
[07:36:51:692] [22397:22398] [ERROR][com.freerdp.core] - freerdp_check_fds() failed - 0
[07:36:51:692] [22397:22398] [INFO][com.freerdp.client.x11] - Network disconnect!
[07:36:51:692] [22397:22398] [ERROR][com.freerdp.client.x11] - Failed to check FreeRDP file descriptor

```

But the server is virtualized in the same machine I'm using to connect, network as NAT and I check connection with ping, and all ping were ok.

Any help?

Thanks

**Comment by akallabeth** [comment]
@rodrigo-yoshioka Would be interesting where exactly the code fails. Could you run this in a debugger? Maybe the exact location of the error can shine some light on that.

**Comment by bmiklautz** [comment]
@rodrigo-yoshioka you are using the old command line syntax. I'd suggest to switch to the new one.
```
xfreerdp /u:Rodrigo /p:PASSWD /cert-ignore /app:'%windir%\system32\calc.exe' /v:192.168.122.133
```

**Comment by akallabeth** [comment]
Fixed with #4459
